pallets/creator-staking/src/tests/tests.rs

```diff
@@ -0,0 +1,2077 @@
+
+        // Ensure that we can unbond up to a limited amount of time.
+        for _ in 0..MAX_UNLOCKING_CHUNKS * 2 {
+            assert_unstake(1, creator_id, 10);
+            assert_eq!(1, BackerLocksByAccount::<TestRuntime>::get(&backer).unbonding_info.len());
+        }
+    })
+}
+
+#[test]
+fn unstake_with_zero_value_is_not_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let creator_id = 1;
+        assert_register(10, creator_id);
+
+        assert_noop!(
+            CreatorStaking::unstake(RuntimeOrigin::signed(1), creator_id, 0),
+            Error::<TestRuntime>::CannotUnstakeZero
+        );
+    })
+}
+
+#[test]
+fn unstake_on_not_operated_creator_is_not_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let creator_id = 1;
+        assert_noop!(
+            CreatorStaking::unstake(RuntimeOrigin::signed(1), creator_id, 100),
+            Error::<TestRuntime>::InactiveCreator
+        );
+    })
+}
+
+#[test]
+fn unstake_too_many_unlocking_chunks_is_not_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let creator_id = 1;
+        assert_register(10, creator_id);
+
+        let backer = 1;
+        let unstake_amount = 10;
+        let stake_amount =
+            MINIMUM_STAKING_AMOUNT * 10 + unstake_amount * MAX_UNLOCKING_CHUNKS as Balance;
+
+        assert_stake(backer, creator_id, stake_amount);
+
+        // Ensure that we can unbond up to a limited amount of time.
+        for _ in 0..MAX_UNLOCKING_CHUNKS {
+            advance_to_era(CreatorStaking::current_era() + 1);
+            assert_unstake(backer, creator_id, unstake_amount);
+        }
+
+        // Ensure that we're at the max but can still add new chunks since it should be merged with the existing one
+        assert_eq!(
+            MAX_UNLOCKING_CHUNKS,
+            CreatorStaking::backer_locks(&backer).unbonding_info.len()
+        );
+        assert_unstake(backer, creator_id, unstake_amount);
+
+        // Ensure that further unbonding attempts result in an error.
+        advance_to_era(CreatorStaking::current_era() + 1);
+        assert_noop!(
+            CreatorStaking::unstake(
+                RuntimeOrigin::signed(backer),
+                creator_id.clone(),
+                unstake_amount
+            ),
+            Error::<TestRuntime>::TooManyUnlockingChunks,
+        );
+    })
+}
+
+#[test]
+fn unstake_on_not_staked_creator_is_not_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let creator_id = 1;
+        assert_register(10, creator_id);
+
+        assert_noop!(
+            CreatorStaking::unstake(RuntimeOrigin::signed(1), creator_id, 10),
+            Error::<TestRuntime>::NotStakedCreator,
+        );
+    })
+}
+
+#[test]
+fn unstake_too_many_era_stakes() {
```

Comment:
I totally didnt understand the test `unstake_too_many_era_stakes`, we are doing `MAX_ERA_STAKE_VALUES` stakes and after this try to do `unstake` and expect error `TooManyEraStakeValues`, why?